app/controllers/api/v2/hosts_bulk_actions_controller.rb

```diff
@@ -63,6 +63,17 @@ def build
         end
       end
 
+      api :PUT, "/hosts/bulk/reassign_hostgroups", N_("Reassign hostgroups")
+      param_group :bulk_host_ids
+      param :hostgroup_id, :number, :required => true, :desc => N_("ID of the hostgroup to reassign the hosts to")
+      def reassign_hostgroup
+        hostgroup = Hostgroup.find(params[:hostgroup_id])
+        BulkHostsManager.new(hosts: @hosts).reassign_hostgroups(hostgroup)
+        process_response(true, { :message => n_("Reassigned %s host to hostgroup %s",
+          "Reassigned %s hosts to hostgroup %s",
+          @hosts.count) % [@hosts.count, hostgroup.name] })
```

Comment:
You can't use multiple `%s` values in a translation string because translators may want to swap the order of words
```suggestion
        process_response(true, { :message => n_("Reassigned {count} host to hostgroup {hostgroup}",
          "Reassigned {count} hosts to hostgroup {hostgroup}",
          @hosts.count) % { count: @hosts.count, hostgroup: hostgroup.name } })
```